tls/s2n_client_hello.c

```diff
@@ -565,6 +565,14 @@ int s2n_client_hello_send(struct s2n_connection *conn)
     struct s2n_stuffer *out = &conn->handshake.io;
     uint8_t client_protocol_version[S2N_TLS_PROTOCOL_VERSION_LEN] = {0};
 
+    /* We don't send version 0x0301
```

Comment:
Should we? Would it cause any performance loss?

At the very least it complicates the code, for a benefit I haven't quantified.